src/Framework/test/PackageTests.cs

```diff
@@ -0,0 +1,139 @@
+                        .GetFiles(packageRoot, "*.nupkg", SearchOption.AllDirectories)
+                        .Where(file => !file.EndsWith(".symbols.nupkg", StringComparison.OrdinalIgnoreCase));
+        if (Directory.Exists(_packageLayoutRoot))
+        {
+            Directory.Delete(_packageLayoutRoot, true);
+        }
+        foreach (var package in packages)
+        {
+            var outputPath = Path.Combine(_packageLayoutRoot, Path.GetFileNameWithoutExtension(package));
+            ZipFile.ExtractToDirectory(package, outputPath);
+        }
+    }
+
+    [Fact]
+    public void PackageAssembliesHaveExpectedAssemblyVersions()
+    {
+        // Do nothing if this is a dev build
+        if (!TestData.VerifyPackageAssemblyVersions() && !SkipOnHelixAttribute.OnHelix())
+        {
+            return;
+        }
+
+        var versionStringWithoutPrereleaseTag = TestData.GetSharedFxVersion().Split('-', 2)[0];
+        var expectedVersion = Version.Parse(versionStringWithoutPrereleaseTag);
+
+        string[] helixTestRunnerToolPackages = { "dotnet-serve", "dotnet-ef", "dotnet-dump" };
+        string[] toolAssembliesToSkip = { "System.", "Microsoft.", "Azure.", "Newtonsoft.", "aspnetcorev2" };
+
+        foreach (var packageDir in Directory.GetDirectories(_packageLayoutRoot))
+        {
+            // Don't test the Shared Framework or Ref pack
+            if (packageDir.Contains("Microsoft.AspNetCore.App", StringComparison.OrdinalIgnoreCase))
+            {
+                continue;
+            }
+            // Don't test helix test runner tool packages
+            if (helixTestRunnerToolPackages.Any(s => packageDir.Contains(s, StringComparison.OrdinalIgnoreCase)))
+            {
+                continue;
+            }
+            // Test lib assemblies
+            var packageAssembliesDir = Path.Combine(packageDir, "lib");
+            if (Directory.Exists(packageAssembliesDir))
+            {
+                foreach (var tfmDir in Directory.GetDirectories(packageAssembliesDir))
+                {
+                    var tfm = new DirectoryInfo(tfmDir).Name;
+                    foreach (var assembly in Directory.GetFiles(tfmDir, "*.dll"))
+                    {
+                        using var fileStream = File.OpenRead(assembly);
+                        using var peReader = new PEReader(fileStream, PEStreamOptions.Default);
+                        var reader = peReader.GetMetadataReader(MetadataReaderOptions.Default);
+                        var assemblyVersion = reader.GetAssemblyDefinition().Version;
+
+                        // net & netstandard assembly versions should all match Major.Minor.0.0
+                        // netfx assembly versions should match Major.Minor.Patch.0
+                        Assert.Equal(expectedVersion.Major, assemblyVersion.Major);
+                        Assert.Equal(expectedVersion.Minor, assemblyVersion.Minor);
+                        if (IsNetFx(tfm))
```

Comment:
nit: No need to evaluate this each time through the `assembly` list. Suggest something like `var isNetFx = IsNetFx(new DirectoryInfo(tfmDir).Name);`. Could also pass `tfmDir` into `IsNetFx(...)` and create the `DirectoryInfo` instance there.